### PR TITLE
MEN-4104: Add D-Bus introspection spec file

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: Documentation/io.mender.Authentication1.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+<node>
+  <!--
+    io.mender.Authentication1:
+    @short_description: Mender Authentication API v1
+
+    This interface lets applications authenticate with the Mender server. The
+    Mender client will handle the authentication and provide to the user with a
+    JSON Web Token (JWT) and the server URL, which which the user can do its own
+    API calls to the Mender server.
```
Comment (tranchitella) [inline Documentation/io.mender.Authentication1.xml:12]: This interface lets applications authenticate with the Mender server. The
Mender client will handle the authentication and provide the user with a
JSON Web Token (JWT) and the server URL, which the user can use to
API calls to the Mender server on his own.

File: Documentation/io.mender.Authentication1.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  -->
+  <interface name="io.mender.Authentication1">
+
+    <!--
+      GetJwtToken:
+      @token: Valid JWT token
+      @server_url: Server URL
+
+      Gets current JWT token and server URL. If no JWT token is available, error
+      is returned.
```
Comment (tranchitella) [inline Documentation/io.mender.Authentication1.xml:22]: Gets current JWT token and server URL. If no JWT token is available, an error
is returned.

File: Documentation/io.mender.Authentication1.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      <arg type="b" name="success" direction="out"/>
+    </method>
+
+    <!--
+      JwtTokenStateChange:
+      @token: Current JWT token
+      @server_url: Server URL
+
+      Emitted whenever a valid JWT is available in the Mender client. The new
+      token is passed with the event, together with the server URL.
```
Comment (tranchitella) [inline Documentation/io.mender.Authentication1.xml:45]: Emitted whenever a valid JWT is available in the Mender client. The event includes the new
token and the server URL.

File: Documentation/io.mender.Authentication1.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      GetJwtToken:
+      @token: Valid JWT token
+      @server_url: Server URL
+
+      Gets current JWT token and server URL. If no JWT token is available, error
+      is returned.
+    -->
+    <method name="GetJwtToken">
+      <arg type="s" name="token" direction="out"/>
+			<arg type="s" name="server_url" direction="out"/>
```
Comment (tranchitella) [inline Documentation/io.mender.Authentication1.xml:26]: Why is the second arg indented with tabs?

File: Documentation/io.mender.Authentication1.xml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      JwtTokenStateChange:
+      @token: Current JWT token
+      @server_url: Server URL
+
+      Emitted whenever a valid JWT is available in the Mender client. The new
+      token is passed with the event, together with the server URL.
+    -->
+    <signal name="JwtTokenStateChange">
+      <arg type="s" name="token"/>
+			<arg type="s" name="server_url"/>
```
Comment (tranchitella) [inline Documentation/io.mender.Authentication1.xml:49]: Why is the second arg indented with tabs?

**Review by tranchitella — COMMENTED** [review]
A few questions and suggestions

**Review by oleorhagen — APPROVED** [review]
No too familiar with the area, but had a read through.

Only thing which struct me is:
Why do we refer to `JWT` tokens in all comments, but the methods all spell it `Jwt`?

File: Documentation/io.mender.Authentication1.xml (hunk 2 above)
Comment (kacf) [reply Documentation/io.mender.Authentication1.xml:22]: Gets current JWT token and server URL. If no JWT token is available, **an empty token**
is returned.

File: Documentation/io.mender.Authentication1.xml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    <method name="GetJwtToken">
+      <arg type="s" name="token" direction="out"/>
+			<arg type="s" name="server_url" direction="out"/>
+    </method>
+
+    <!--
+      FetchJwtToken:
+      @success: false on errors
+
+      Instructs the Mender client to fetch the JWT token from the server.
```
Comment (kacf) [inline Documentation/io.mender.Authentication1.xml:33]: Append: "When the token is ready, a FetchJwtToken signal will be sent".

File: Documentation/io.mender.Authentication1.xml (hunk 4 above)
Comment (lluiscampos) [reply Documentation/io.mender.Authentication1.xml:26]: My editor playing tricks on me

**Comment by lluiscampos** [comment]
> Only thing which struct me is:
> Why do we refer to `JWT` tokens in all comments, but the methods all spell it `Jwt`?

We did name our methods `Jwt`, we cannot do anything about that now, while `JWT` is the generally used name (just Google-ing a bit).

**Comment by lluiscampos** [comment]
@tranchitella @kacf Thank you for your feedback. I rewrote it following your suggestions.

**Review by tranchitella — APPROVED** [review]

**Pull request merged**